frontend/tests/features/download.test.ts

```diff
@@ -0,0 +1,83 @@
+import { expect, test } from "@playwright/test";
+import {
+  deleteCsvFile,
+  downloadCsv,
+  getCsvMetadata,
+  goToWMG,
+  selectFilterOption,
+  selectTissueAndGeneOption,
+  verifyMetadata,
+} from "../utils/wmgUtils";
+import { isDevStagingProd } from "tests/utils/helpers";
+
+const EXPECTED_HEADER = [
+  "Tissue",
+  "Cell Type",
+  "Cell Count",
+  "Tissue Composition",
+  "Gene Symbol",
+  "Expression",
+  '"Expression',
+  ' Scaled"',
+  "Number of Cells Expressing Genes",
+];
+const BLOOD_TISSUE_COUNT =
+  '[data-testid="cell-type-labels-blood"] [data-testid="cell-type-label-count"]';
+const { describe, skip } = test;
+describe("Csv download", () => {
+  skip(!isDevStagingProd, "WMG BE API does not work locally or in rdev");
+  test.beforeEach(async ({ page }) => {
+    // navigate to gene expression page
+    await goToWMG(page);
+    //select tissue and gene
+    await selectTissueAndGeneOption(page);
+  });
+  [
+    ["dataset-filter"],
```

Comment:
Instead of this, move the test body to reusable function and break this into individual tests. This way your wmgUtil methods will not have to be that big